src/Workspaces/Core/Portable/Execution/CustomAssetBuilder.cs

```diff
@@ -23,16 +23,14 @@ public CustomAsset Build(OptionSet options, string language, CancellationToken c
         {
             cancellationToken.ThrowIfCancellationRequested();
 
-            return new SimpleCustomAsset(WellKnownSynchronizationKinds.OptionSet, 
-                (writer, cancellationTokenOnStreamWriting) => 
+            return new SimpleCustomAsset(WellKnownSynchronizationKinds.OptionSet,
+                (writer, cancellationTokenOnStreamWriting) =>
                     _serializer.SerializeOptionSet(options, language, writer, cancellationTokenOnStreamWriting));
         }
 
         public CustomAsset Build(AnalyzerReference reference, CancellationToken cancellationToken)
         {
-            return new SimpleCustomAsset(WellKnownSynchronizationKinds.AnalyzerReference, 
-                (writer, cancellationTokenOnStreamWriting) => 
-                    _serializer.SerializeAnalyzerReference(reference, writer, cancellationTokenOnStreamWriting));
+            return new WorkspaceAnalyzerReferenceAsset(reference, _serializer);
```

Comment:
I don't understand how this change relates to the change in `VisualStudioDiagnosticAnalyzerExecutor`.